### PR TITLE
Reduce running time of TestAutodeleteOwnerRefs StatefulSet Integration Tests

### PR DESCRIPTION
**Pull request by pwschuurman**
Move expensive test setup (apiserver and running controller) to be run once in TestAutodeleteOwnerRefs parameterized tests

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Moves the expensive setup of the TestAutodeleteOwnerRefs to be run once, and each parameterized test can runs in its own namespace for isolation. Runtime reduces from ~15s -> ~5s (when testing locally).

```
go test ./test/integration/statefulset -run 'TestAutodeleteOwnerRefs' 
ok  	k8s.io/kubernetes/test/integration/statefulset	5.820s
```

#### Special notes for your reviewer:
Inspired from this comment: https://github.com/kubernetes/kubernetes/pull/115466#discussion_r1094483349

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: test/integration/statefulset/statefulset_test.go [hunk 1]
```diff
@@ -397,12 +397,14 @@ func TestStatefulSetStatusWithPodFail(t *testing.T) {
 func TestAutodeleteOwnerRefs(t *testing.T) {
 	tests := []struct {
 		name              string
+		namespace         string
```
Comment (mattcary) [inline test/integration/statefulset/statefulset_test.go:400]: To reduce toil and repetitive boilerplate we could probably drop the "name" field in the struct and just use the namespace for documentation?

**Review by mattcary — COMMENTED** [review]
Nice find, looking through the existing tests I see there's a big mix between doing this setup per-test case or per-test.

One minor nit.

File: test/integration/statefulset/statefulset_test.go (hunk 1 above)
Comment (pwschuurman) [reply test/integration/statefulset/statefulset_test.go:399]: Done.

**Comment by mattcary** [comment]
Ooops, dropped this.

Still lgtm, will need rebase.

**Comment by pwschuurman** [comment]
> Ooops, dropped this.
> 
> Still lgtm, will need rebase.

Rebased and updated.

**Pull request merged**